### ISSUE TITLE
a11y test on `Badge`

### ISSUE THREAD

**Issue by Barsnes** [digdir/designsystemet#3560]
We are currently unsure about what a11y tips to give to consumers.

Currently `badge`

**Comment by Barsnes** [comment]
>Currently badge won't give any info to screenreaders, since they are rendered using only CSS.

**This is false**, at least for NVDA in Chrome on windows
<img width="1100" height="787" alt="Image" src="https://github.com/user-attachments/assets/4a990e62-aa9c-478d-9bbf-00bde1e6c5d7" />

**Comment by Camulos** [comment]
VoiceOver + Edge:
"99+ Innboks, knapp, gruppe"
"10 Meldinger, knapp, gruppe"
VoiceOver + Safari:
"99+ Innboks, knapp"
"10 Meldinger, knapp"

**Comment by 42tte** [comment]
Elementet går inte att markera och få uppläst i Safari+VoiceOver (macOS), fungerar i Chrome+VoiceOver

**Comment by sigveres** [comment]
Undersøk om det kan være bedre å bruke `aria-describedby` som peker til en span med sr-only klasse hvor nummeret står i ren tekst. Da vil nummeret av meldinger være tilleggsinformasjon, og ikke en del av labelen. Når man brukere `aria-labelledby` endres tilgjengelighetsnavnet som kan gi konflikt med [WCAG 2.5.3 - ledetekst i navn](https://www.uutilsynet.no/wcag-standarden/253-ledetekst-i-navn-niva/150) man står også litt friere dersom man ønsker å legge til tilleggsinformasjon utover data-verdien.

På den måten kan brukere med tale-navigering si: "Innboks", ikke "99+ Innboks" for å navigere/lese opp elementet

**Eksempel**: 
En liten test direkte i HTMLen hvor jeg la inn `aria-describedby` på `<button>` som peker til en span som er skjult. 
<img width="960" height="617" alt="Image" src="https://github.com/user-attachments/assets/2698fde5-d0f1-4cf3-a61c-0023c8dd411f" />

Bildet accessibility viewet i nettleser hvor det fremheves at navnet for knapp med ` aria-labelledyby ` er "99+ Innboks" mens `aria-describedby` står det "Innboks"
<img width="741" height="522" alt="Image" src="https://github.com/user-attachments/assets/38779d9b-8714-4d4e-9997-21ead016cd1d" />

**Comment by Febakke** [comment]
Tested with VoiceOver/chrome

Did some testing now and I think we need to be more specific in how consumers implement this. I agree with @sigveres that `aria-describeby` is the best option when badge is used with interactive elements. 

First of there is a difference if the element is used inline or around the element. If its inline the number will be a part of the accessible name of the button and read by VoiceOver. If its outside, it has nothing to do with the button and will not be read. In my opinion it would be better to hide the badge from screen readers and give the extra info on the `button` element instead. For our users I think it would be better to have one consistent way of doing it instead of having different approaches depending on how you use `badge`. 

## Summary of VoiceOver tests
### `Aria-Label` on `badge`
<img width="1169" height="288" alt="Inline: Screen reader reads the badge aria-label when the badge is placed inline" src="https://github.com/user-attachments/assets/85427e96-d206-4169-9da1-5ada73c709c3" />

<img width="1188" height="336" alt="Position: Screen reader do not read the badge aria-lable when the badge is positioned" src="https://github.com/user-attachments/assets/2dbd9ba2-d216-4fc9-befa-652df4a349a5" />

Aria-label do not fix the issue, its only read with button if its inline. This is the same behavior as with a "vanilla" badge. Where "2" in this case would be read "inline" but not when positioned. 

### `Aria-describedby` on `Button`
<img width="1165" height="528" alt="Inline: Screen reader reads button text, number inside badge and aria-describedby" src="https://github.com/user-attachments/assets/0c80fae5-d562-441f-a7b2-974e6098f615" />

<img width="1208" height="535" alt="Position: Screen reader reads button text and aria-describedby" src="https://github.com/user-attachments/assets/b05c1a8f-033f-479b-b5f1-2a785062137b" />

As you can see, its more consistent. But we get the number from badge read when its placed inline. 

### `Aria-describedby` on `Button` and `aria-hidden` on Badge

<img width="1125" height="802" alt="Inline: Screen reader reads Button text and aria-describedby" src="https://github.com/user-attachments/assets/25e3f497-adf0-4326-9a8e-1e5c86ecda8c" />

<img width="1180" height="741" alt="Position: Screen reader reads Button text and aria-describedby" src="https://github.com/user-attachments/assets/c81145ec-e584-4852-9baf-bee84c724499" />

This is the most consistent variant in my opinion. A plus side to this approach is that you can add extra info if "2" is not enough. You could say "2 new messages" for instance. 

Im guessing its hard to do this "automaticly" with our `badge` component but we should document how to set this up correctly.

**Comment by Febakke** [comment]
One more accessibilty aspect to look at: What about when a `Badge` is dynamic. For instance its added to you inbox icon when a new message has been recieved. Something to look at for our documentation as well.

**Comment by Febakke** [comment]
Summary after discussion with @eirikbacker 
- We should avoid using `aria-describedby` because it will not be visible in VoiceOver rotor and we are guessing similar behavior in other screen readers.
- We discussed a solution that would add `badge` as a property to relevant components. For instance `Avatar` and `Button`. But decided this would not work as `Badge` might be used as a static standalone element as well. 
- The best way forward is to avoid examples and documentation telling users to add `badge` outside the interactive element. If the badge is a part of the text element in a button for instance it will be read as a part of the label. This is the best solution for screen readers. In this solution you could also use `aria-label` on the badge to give more information then the content might give you and have it be part of the `Button` text. 

## Tasks
- [ ] Investigate: Can we just recommend using `Badge` "inline"?
  - [ ] Specify that this is important on interactive components
  - [ ] Create examples on how it could look inline ([Inspiration from Mattilsynet)](https://design.mattilsynet.no/?path=/story/designsystem-badge--with-position)
- [ ] Test if adding `badge` inline on relevant components work well with screen reader (Both VoiceOver and NVDA) 
  - [ ] Button
  - [ ] Togglegroup
  - [ ] Clickable Avatar example
  - [ ] Tabs
  - [ ] Elements in dropdown
  - [ ] Other?
- [ ] #4286 Should have implementation test for Badge on how it works with screen reader

**Comment by mrosvik** [comment]
This is now documented in the Badge Accessibility docs.
